resources/lemmas.md

```diff
@@ -72,6 +82,13 @@ They capture the essential mechanisms used by the two instructions: splitting a
                 : nthbyteof(V, 31, 32)
                 : .WordStack ) => V
       requires 0 <=Int V andBool V <Int pow256
+
+    rule 0 <=Int #asWord(#bufSeg(_, _, _))                 => true
+    rule         #asWord(#bufSeg(_, _, WIDTH)) <Int pow256 => true  requires WIDTH <=Int 32
+
+    rule         #asWord(#bufSeg(_, _, 1)) <=Int 255 => true
+
+    rule #buf(32, #asWord(#bufSeg(BUF, START, WIDTH))) => #bufSeg(BUF, START +Int WIDTH -Int 32, 32)  requires WIDTH >=Int 32
```

Comment:
i'm surprised here. is this really happens in the bytecode? could you elaborate the scenario?

Reply:
This is a more general rule. The actual instance used in proving is when WIDTH ==Int 32. rhs is the actual representation that appears during the execution. lhs is the result of `#bytes32(#asWord(#bufSeg(#buf(LEN, SIGS), 65 *Int POS        , 32)))` (which becomes `#buf(32, #asWord(#bufSeg(#buf(LEN, SIGS), 65 *Int POS        , 32)))`

Reply:
to be more general, i think we need to define the case when WIDTH < 32?